plugins/notifications/src/components/NotificationsSideBarItem/NotificationsSideBarItem.tsx

```diff
@@ -259,6 +274,9 @@ export const NotificationsSidebarItem = (props?: {
       )}
       <SidebarItem
         to={notificationsRoute()}
+        onClick={() => {
+          requestUserPermission();
```

Comment:
This is a good change to make it working in all (?) browsers.
On the other hand, the browser notifications will be shown after visiting the page only. This can be confusing.

What about keeping that for the sake of compatibility here and having the former `useEffect()` as well? If the browser does not enforce an user action to check permission, the user experience will be better. The only drawback is a `[Violation] ...` message in the log without consequences.

Reply:
I think this is more future-proof if they decide that it is required to have the user interaction and will not work the other way at all. I don't think it's confusing as it will ask the permission once you go to the notifications page the first time and never after that. Other option would be to have some sort of banner/modal to ask for user permission on page load but that requires more work that can be separated to own PR if we decide to do that. 

What do you think?

Reply:
I am for the most intrusive approach as possible, so no banners, etc. :-)

If we keep recent implementation, we will get uniform behavior across browsers, which is probably better. So let's keep it as it is.